### PR TITLE
feat: windows support for toolchain

### LINKED ISSUE #1000: Windows support for Wing CLI
As a user with a Windows machine, I would like to be able to use Wing.

Our CLI used to work with Windows before, so we need to:
- [x] Fix the CLI, so it works on Windows machines
- [x] Add a test to our CI to continuously build on a Windows machine (and break when it doesn't)

Related:
#644 
#630

### PR DESCRIPTION
**Pull request by MarkMcCulloh**
- Undoes some changes from previous PRs regarding windows, ideally tries to do the _path normalization_ as infrequently and late as necessary
- Rewrite of hangar to no longer use https://github.com/google/zx. It had issues with Windows and/or Vitest. It was useful before when hangar was more complicated, but now just using https://github.com/sindresorhus/execa instead
  - hangar is functionally the same

Fixes #1000

### CODE REVIEW

**Review by eladb — APPROVED** [review]
Ship it yeah!

File: apps/wing/src/wingc.ts [hunk 1]
```diff
@@ -24,19 +24,27 @@ export interface WingCompilerLoadOptions {
...
+    // .jsii access
+    [WINGSDK_MANIFEST_ROOT]: WINGSDK_MANIFEST_ROOT,
+    ...(options.preopens ?? {}),
+  } as Record<string, string>;
+
+  if (process.platform === "win32") {
+    for (const [key, value] of Object.entries(preopens)) {
+      delete preopens[key];
+      preopens[normalPath(value)] = value;
+    }
```
Comment (Chriscbr) [inline apps/wing/src/wingc.ts:37]: Mutating the same collection you're iterating over.... living life dangerously I see? 😄

Comment (MarkMcCulloh) [reply apps/wing/src/wingc.ts:37]: 😎 
Technically iterating over `Object.entries(preopens)`, so should be completely safe to mutate the original afaik

**Comment by revitalbarletz** [comment]
Does it mean that win developers can now contribute as well?

**Comment by staycoolcall911** [comment]
Generally yes, but wing console is not supported yet on windows, see [Wing Console Windows support](https://github.com/winglang/wing/issues/630)

**Comment by MarkMcCulloh** [comment]
@revitalbarletz If you mean can windows developers build the wing repo? That is untested and I don't think we've talked about ever supporting that

**Comment by revitalbarletz** [comment]
@MarkMcCulloh can you please elaborate what is included in this change?

**Comment by MarkMcCulloh** [comment]
@revitalbarletz This PR adds a couple of fixes for Windows and includes it in our E2E testing. This is specifically for running Wing as a user, e.g. running `wing compile -t sim cool.w` on Windows in a windows-native shell like PowerShell.

However:
- Someone using Windows is not well supported to develop/contribute to this project. They may be able to get everything working, but it's not an officially considered development environment
- The Wing Console is not directly affected by the changes here and requires its own work to support Windows in any way

**Pull request merged**